projects/igniteui-angular-elements/src/analyzer/elements.config.ts

```diff
@@ -16,11 +17,12 @@ export const registerComponents = [
 ];
 //// WARNING: Code below this line is auto-generated and any modifications will be overwritten
 export var registerConfig = [
-    { component: IgxColumnComponent, parents: [IgxGridComponent, IgxTreeGridComponent, IgxHierarchicalGridComponent, IgxPivotGridComponent, IgxRowIslandComponent], contentQueries: [], methods: ["getInitialChildColumnSizes", "getFilledChildColumnSizes", "getResizableColUnderEnd", "pin", "unpin", "move", "autosize"] },
+    { component: IgxColumnGroupComponent, parents: [IgxGridComponent, IgxTreeGridComponent, IgxHierarchicalGridComponent, IgxColumnGroupComponent, IgxRowIslandComponent], contentQueries: [{ property: "children", childType: IgxColumnComponent, isQueryList: true }], methods: ["getInitialChildColumnSizes", "getFilledChildColumnSizes", "getResizableColUnderEnd", "pin", "unpin", "move", "autosize"] },
+    { component: IgxColumnComponent, parents: [IgxGridComponent, IgxTreeGridComponent, IgxHierarchicalGridComponent, IgxPivotGridComponent, IgxRowIslandComponent, IgxColumnGroupComponent], contentQueries: [], methods: ["getInitialChildColumnSizes", "getFilledChildColumnSizes", "getResizableColUnderEnd", "pin", "unpin", "move", "autosize"] },
     { component: IgxGridComponent, parents: [], contentQueries: [{ property: "columnList", childType: IgxColumnComponent, isQueryList: true }, { property: "toolbar", childType: IgxGridToolbarComponent, isQueryList: true }, { property: "paginationComponents", childType: IgxPaginatorComponent, isQueryList: true }], methods: ["getCellByColumnVisibleIndex", "groupBy", "clearGrouping", "isExpandedGroup", "toggleGroup", "selectRowsInGroup", "deselectRowsInGroup", "fullyExpandGroup", "toggleAllGroupRows", "getSelectedData", "getRowByIndex", "getRowByKey", "getCellByColumn", "getCellByKey", "pinRow", "unpinRow", "isRecordPinnedByIndex", "toggleColumnVisibility", "expandAll", "collapseAll", "expandRow", "collapseRow", "toggleRow", "getHeaderGroupWidth", "getColumnByName", "getColumnByVisibleIndex", "nextPage", "previousPage", "moveColumn", "paginate", "markForCheck", "addRow", "deleteRow", "updateCell", "updateRow", "getRowData", "sort", "filter", "filterGlobal", "enableSummaries", "disableSummaries", "clearFilter", "clearSort", "pinColumn", "unpinColumn", "reflow", "findNext", "findPrev", "refreshSearch", "clearSearch", "getPinnedWidth", "selectRows", "deselectRows", "selectAllRows", "deselectAllRows", "selectedColumns", "selectColumns", "deselectColumns", "deselectAllColumns", "selectAllColumns", "getSelectedColumnsData", "combineSelectedCellAndColumnData", "navigateTo", "getNextCell", "getPreviousCell", "openAdvancedFilteringDialog", "closeAdvancedFilteringDialog", "endEdit", "beginAddRowById", "beginAddRowByIndex"] },
```

Comment:
Hm, fairly certain this isn't picking up the content queries for the parents (since the query is for the column base class and the group overrides it). Does it actually work if the grid isn't notified of the groups? I'd assume there would be a case where something would break due to that

Reply:
Oh just saw the change in the strategy. Okay, I guess that'll do, though strictly speaking the query won't load subclass components without them explicitly providing themselves as the concrete query type. Wondering if there are cases where components look for such components separately, though for now we could just leave a comment to look into this down the line